src/GraphQLController.php

```diff
@@ -29,7 +29,7 @@ public function query(Request $request, string $schema = null): JsonResponse
         // If there are multiple route params we can expect that there
         // will be a schema name that has to be built
         $routeParameters = $this->getRouteParameters($request);
-        if (count($routeParameters) > 1) {
+        if (count($routeParameters) > 0) {
```

Comment:
I don't think the fix is correct here and should rather be within `\Rebing\GraphQL\GraphQLController::getRouteParameters`, can you investigate?

Reply:
> 
> 
> I don't think the fix is correct here and should rather be within `\Rebing\GraphQL\GraphQLController::getRouteParameters`, can you investigate?

Thanks mfn for having a look at my PR :) I don't see what can be fixed in the getRouteParameters as it already handle the Lumen case, and return the only one element in the route param available ..

Reply:
Yes, so my impression is "if it's broken with Lumen, it need to be fixed in the Lumen code path".

This change here affects _all_ environments == also Laravel, so I'm not confident this is correct (as no one reported problems with Laravel).

Could I explain my position / concerns?

thanks

Reply:
Ok I get your point.

The situation here is kind of tricky because with Lumen and only one element in the route URL, the schema is never assigned, and the default schema will always be picked.

Could I write something like this instead : 

```php
$routeParameters = $this->getRouteParameters($request);
if (count($routeParameters) > Helpers::isLumen() ? 0 : 1) {
        $schema = implode('/', $routeParameters);
}
```
This fix wouldn't be necessary if we could pass a schema name directly into function arg : 
` public function query(Request $request, string $schema = null): JsonResponse {}`

Reply:
> if (count($routeParameters) > Helpers::isLumen() ? 0 : 1) {

But you can to the same fix _within_ `getRouteParameters` in a lumen specific code path, no?

> This fix wouldn't be necessary if we could pass a schema name directly into function arg :

That would probably require adapting the `routes.php` => I'm open to that, but would see some concrete code.

Also we would definitely need a test for this.

Reply:
Sorry I'm on two parallel tasks, and I'm fairly new to PHP so I may be annoying. But my feeling here is that the `count($routeParameters) > 1` is not right, why would we want to take default schema, if there is one path element anyway ? it still seems valid.

Unfortunatly I don't know how it's supposed to behave with Laravel (what are contained in the `route()->parameters`; but my guess it that it sends the full path as an array in the $parameters value so for instance accessing the 'private' graphql schema would return : 
array('graphql', 'private') when Lumen is returning array('private'). So I guess we could just check if there is a 'graphql' element in the array, and remove it, and treat all values from getRouteParameters in the same way.

I'll update the PR to reflect that and try to update tests

Reply:
Ok I took some time again to look at the code, and I finally found something amazing 😲 

Into **graphql.php** config file, if you replace the academic `{graphql_schema?}` with `{schema}` in Lumen, it works so hard that I can't even think about how many hours I lost trying to wrap my head around it. I spotted the good place to log into RoutesRequests.php file from Lumen Framework, and I saw that the pattern given as a config key '**routes**' in graphql.php was used to insert a into the parameters the schema with this form (e.g private schema ) : 
`["private": "private"]`

Now if we change the config key to `{schema}` it will then automatically add into parameters : 
`["schema": "private"]`

Hence, the GraphQLController gets hit with correct schema value.

🗡 🗡 I could easily do Sepukku, if we were 600 years back in the past

Jokes aside, I will close this non-sense PR as it works with Lumen, BUT I will open another PR to update documentation about that.